lib/bin/build_env.sh

```diff
@@ -29,6 +35,23 @@ elif [ -n "$SEMAPHORE" ]; then
   run_attempt=1
   runner_id=$SEMAPHORE_JOB_ID
   pr_title=$SEMAPHORE_GIT_PR_NAME
+  commit_message=$(git log --format=%s -n 1 $sha)
+  committer_name=$(git show -s --format='%an' -n 1 $sha)
+  committer_email=$(git show -s --format='%ae' -n 1 $sha)
+elif [ -n "$MINT" ]; then
+  platform=mint
+  branch="${MINT_GIT_REF_NAME}"
+  actor="${MINT_ACTOR}"
+  sha="${MINT_GIT_COMMIT_SHA}"
+  run_id="${MINT_RUN_ID}"
+  run_attempt="${MINT_TASK_ATTEMPT_NUMBER}"
+  runner_id="${MINT_PARALLEL_INDEX}"
+  # Mint does not preserve the .git directory by default to improve the likelihood of cache hits. Instead
+  # of asking git for commit information, then, we rely on the mint/git-clone leaf to populate the necessary
+  # metadata in environment variables.
+  commit_message="${MINT_GIT_COMMIT_SUMMARY}"
+  committer_name="${MINT_GIT_COMMITTER_NAME}"
+  committer_email="${MINT_GIT_COMMITTER_EMAIL}"
```

Comment:
I tried to mimic the existing behavior w.r.t. the format of the git commands you are feeding to message/name/email. 

Technically, we do also expose `MINT_GIT_COMMIT_MESSAGE` but it's the full body, not just the summary line.

Reply:
See also: https://github.com/rwx-research/mint-leaves/blob/1ef47f8108776c055f78941e6cbe30b6253af690/mint/git-clone/mint-leaf.yml#L100-L108